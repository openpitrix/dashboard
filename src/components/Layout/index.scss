@import '~scss/vars';

.layout {
  padding-top: $layout-top;
  min-height: $content-height;
  width: $content-width;
  margin: 0 auto;

  &.hasBack {
    padding-top: $layout-top - 20px;
  }

  &.hasMenu,
  &.hasNav {
    margin: 56px 20px 32px $menu-width + 20px;
    //min-width: 950px;
    min-width: 992px;
    min-height: calc(100vh - 100px);
    padding-top: 0;
  }

  &.hasMenu {
    width: calc(100% - #{$menu-width} - 40px);
  }

  &.hasNav {
    width: calc(100% - #{$menu-nav-width} - 40px);
    margin-left: $menu-nav-width + 20px;
  }

  &.detailPage {
    width: calc(100% - #{$menu-nav-width});
    margin-left: $menu-nav-width;
    margin-top: 88px;
  }

  .backBtn {
    //width: $content-width;
    margin: 0 auto 20px;
    font-size: $size-normal;
    line-height: 1.43;
    a {
      color: $P75;
    }
  }

  &.noTabs {
    padding-top: $header-height + 32px;
  }

  .detailTab {
    position: absolute;
    top: 150px;
    left: 384px;
    background-color: #fff;
    & > div {
      top: 100px;
    }
  }

  .socketMessage {
    position: fixed;
    top: 70px;
    left: 88%;
  }
}

.titleBanner {
  position: fixed;
  top: 48px;
  left: 0;
  z-index: 3;
  width: 100%;
  height: $title-banner-height;
  background-color: $N500;

  .wrapper {
    width: $content-width;
    margin: 30px auto;
  }

  .image,
  .icon {
    float: left;
    margin-right: 12px;
    width: 48px;
    height: 48px;
  }

  .image {
    line-height: 48px;

    img {
      max-width: 100%;
      max-height: 100%;
      vertical-align: middle;
    }
  }

  .icon {
    box-sizing: border-box;
    padding: 12px;
    background-color: $N400;
    border-radius: 3px;
  }

  .name {
    margin-bottom: 4px;
    font-size: 20px;
    font-weight: 500;
    line-height: 24px;
    letter-spacing: 0;
    color: $N0;
  }

  .desc {
    height: 20px;
    line-height: 20px;
    font-size: 12px;
    color: $N65;
    overflow: hidden;
  }

  .search {
    float: right;
    width: 360px;
    i {
      color: $N65;
    }
    &:hover {
      i {
        color: $P100;
      }
    }
    input {
      color: $N500;
      padding-left: 36px;
      border-radius: 16px;
      border: 1px solid transparent;
      background-color: $N10;
      transition: all 0.3s ease-in-out;
      &::placeholder {
        font-size: 14px;
        color: $N45;
        line-height: 1;
        letter-spacing: 0;
      }
      &:focus {
        border: solid 1px $P75 !important;
      }

      &:hover {
        border: solid 1px $N45;
      }
    }
    :global {
      .icon {
        color: $N65;
        opacity: 0.5;
        &:hover,
        &:focus {
          opacity: 1;
        }
      }
    }
  }
}

.pageTitle {
  position: fixed;
  left: $menu-nav-width;
  top: 0;
  z-index: 2;
  width: 100%;
  height: 64px;
  box-shadow: inset 0 -1px 0 0 $N10;
  background-color: $N0;

  .title {
    margin: 0 auto;
    width: calc(100% - #{$min-center-page-margin}*2);
    max-width: $max-center-page-width;
    line-height: 64px;
    font-size: 16px;
    font-weight: 500;
    color: $N500;
  }

  .icon {
    float: left;
    margin-top: 22px;
    margin-left: -40px;
    line-height: 20px;
    cursor: pointer;
  }
}

.centerPage {
  margin: 0 auto;
  width: calc(100% - #{$min-center-page-margin}*2);
  max-width: $max-center-page-width;
}
