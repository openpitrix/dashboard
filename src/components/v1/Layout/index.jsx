import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import styles from './index.scss';

const Layout = ({ className, children }) => {
  return <div className={classnames(styles.container, className)}>{children}</div>;
};

Layout.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node
};

Layout.defaultProps = {
  className: ''
};

export Section from './Section';
export default Layout;
