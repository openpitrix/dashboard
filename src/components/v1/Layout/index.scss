@import "~scss/vars";

.container {
  display: flex;
  width: $content-width;
  margin: 0 auto;
  justify-content: flex-start;
}