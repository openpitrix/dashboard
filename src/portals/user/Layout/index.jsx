import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Header from 'components/Header';
import Footer from 'components/Footer';

import styles from './index.scss';

export default class UserPortalLayout extends React.PureComponent {
  static propTypes = {
    banner: PropTypes.node,
    className: PropTypes.string
  };

  static defaultProps = {
    banner: null
  };

  render() {
    const { children, banner, className } = this.props;

    return (
      <Fragment>
        <Header />
        {banner}
        <div className={classnames(styles.page, className)}>{children}</div>
        <Footer />
      </Fragment>
    );
  }
}
