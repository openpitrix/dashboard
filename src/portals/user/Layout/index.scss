@import "~scss/vars";

$page-width: 1280px;
$offset-top: $header-height + 30 *2 + 52;

.page {
  position: relative;
  top: $offset-top;
  width: $page-width;
  margin: 36px auto 80px auto;
}
