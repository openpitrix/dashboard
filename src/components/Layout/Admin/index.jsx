import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { inject } from 'mobx-react';
import Notification from 'components/Base/Notification';
import { Container } from 'components/Layout';
import TabsNav from 'components/TabsNav';
import Loading from 'components/Loading';
import { getSessInfo } from 'src/utils';

import styles from './index.scss';

@inject('sessInfo')
export default class Layout extends React.Component {
  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.node,
    msg: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
    hideMsg: PropTypes.func,
    noTabs: PropTypes.bool,
    noNotification: PropTypes.bool,
    isLoading: PropTypes.bool,
    loadClass: PropTypes.string
  };

  static defaultProps = {
    msg: '',
    hideMsg: () => {},
    noTabs: false,
    noNotification: false
  };

  constructor(props) {
    super(props);
    this.availableLinks = [];
    this.linkPrefix = '/dashboard';
  }

  renderNotification() {
    if (this.props.noNotification) return null;
    let { msg } = this.props;
    if (typeof msg === 'object') {
      msg = msg + ''; // transform mobx object
    }
    const { hideMsg } = this.props;

    return msg ? <Notification message={msg} onHide={hideMsg} /> : null;
  }

  renderTabs() {
    const loginRole = getSessInfo('role', this.props.sessInfo);
    const normalLinks = [{ '': 'overview' }, 'apps', 'clusters', 'runtimes'];

    if (loginRole === 'normal') {
      this.availableLinks = [...normalLinks];
    } else if (loginRole === 'developer') {
      this.availableLinks = [...normalLinks, 'repos'];
    } else if (loginRole === 'admin') {
      this.availableLinks = [...normalLinks, 'repos', 'users', 'categories'];
    }

    const options = { prefix: this.linkPrefix };

    return <TabsNav links={this.availableLinks} options={options} />;
  }

  render() {
    const { className, noTabs, noNotification, children, isLoading, loadClass } = this.props;
    return (
      <div className={classnames(styles.layout, className, { [styles.noTabs]: noTabs })}>
        {noTabs ? null : this.renderTabs()}
        {noNotification ? null : this.renderNotification()}
        <Container>
          <Loading isLoading={isLoading} className={styles[loadClass]}>
            {children}
          </Loading>
        </Container>
      </div>
    );
  }
}

export BackBtn from './BackBtn';
export Dialog from './Dialog';
export CreateResource from './CreateResource';
