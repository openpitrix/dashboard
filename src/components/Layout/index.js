import Layout from './index.jsx';

export CreateResource from './CreateResource';
export Dialog from './Dialog';
export BackBtn from './BackBtn';

export Grid from './Grid';
export Section from './Grid/section';
export Row from './Grid/row';
export Panel from './Panel';
export Card from './Card';

export default Layout;
