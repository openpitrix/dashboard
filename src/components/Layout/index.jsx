import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import classnames from 'classnames';
import { inject } from 'mobx-react';
import {
  noop, clone, isEmpty, get
} from 'lodash';

import { Notification, Icon } from 'components/Base';
import Loading from 'components/Loading';
import { getScrollTop } from 'src/utils';
import TitleBanner from './TitleBanner';
import SideNav from './SideNav';

import styles from './index.scss';

@inject(({ rootStore, sock }) => ({
  user: rootStore.user,
  sock
}))
class Layout extends Component {
  static propTypes = {
    backBtn: PropTypes.node,
    children: PropTypes.node,
    className: PropTypes.string,
    hasSearch: PropTypes.bool,
    isHome: PropTypes.bool,
    isLoading: PropTypes.bool,
    listenToJob: PropTypes.func,
    loadClass: PropTypes.string,
    noNotification: PropTypes.bool,
    pageTitle: PropTypes.string,
    title: PropTypes.string
  };

  static defaultProps = {
    noNotification: false,
    backBtn: null,
    listenToJob: noop,
    title: '',
    pageTitle: '',
    hasSearch: false,
    isHome: false
  };

  componentDidMount() {
    const { sock, listenToJob } = this.props;

    sock
      && sock.on('ops-resource', (payload = {}) => {
        const { type } = payload;
        const { resource = {} } = payload;

        listenToJob({
          op: `${type}:${resource.rtype}`,
          type,
          ...resource
        });
      });
  }

  componentWillUnmount() {
    const { sock } = this.props;

    if (sock && !isEmpty(sock._events)) {
      sock._events = {};
    }
  }

  goBack = () => {
    const { history } = this.props;
    history.goBack();
  };

  render() {
    const {
      className,
      noNotification,
      children,
      isLoading,
      loadClass,
      backBtn,
      hasSearch,
      title,
      isHome,
      match,
      pageTitle
    } = this.props;

    const { isNormal, isDev, isAdmin } = this.props.user;
    const hasMenu = (isDev || isAdmin) && !isHome;
    const paths = ['/dashboard', '/profile', '/ssh_keys', '/dev/apps'];
    const isCenterPage = Boolean(pageTitle); // detail page, only one level menu
    const hasSubNav = hasMenu && !isCenterPage && !paths.includes(match.path);

    return (
      <div
        className={classnames(
          styles.layout,
          className,
          { [styles.hasMenu]: hasSubNav },
          { [styles.hasNav]: hasMenu && !hasSubNav },
          { [styles.hasBack]: Boolean(backBtn) },
          { [styles.detailPage]: isCenterPage }
        )}
      >
        {noNotification ? null : <Notification />}
        {backBtn}

        {hasMenu && <SideNav hasSubNav={hasSubNav} />}
        {isNormal
          && !isHome && <TitleBanner title={title} hasSearch={hasSearch} />}

        {isCenterPage && (
          <div className={styles.pageTitle}>
            <div className={styles.title}>
              <Icon
                onClick={this.goBack}
                name="previous"
                size={20}
                type="dark"
                className={styles.icon}
              />
              {pageTitle}
            </div>
          </div>
        )}

        <Loading isLoading={isLoading} className={styles[loadClass]}>
          {isCenterPage ? (
            <div className={styles.centerPage}>{children}</div>
          ) : (
            children
          )}
        </Loading>
      </div>
    );
  }
}

export default withRouter(Layout);
