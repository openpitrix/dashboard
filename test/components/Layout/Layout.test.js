import { Provider } from 'mobx-react';
import { BrowserRouter } from 'react-router-dom';
import { observable } from 'mobx';

import Layout from 'components/Layout';

const renderPage = (store, props) => (
  <Provider rootStore={store}>
    <BrowserRouter>
      <Layout {...props}>
        <div>test</div>
      </Layout>
    </BrowserRouter>
  </Provider>
);

describe('Layout', () => {
  let store;
  let props = {
    noNotification: true
  };
  beforeAll(() => {
    store = observable.object({
      user: {}
    });

    props = {
      ...props
    };
  });

  it('basic render', () => {
    const wrapper = render(renderPage(store, props));

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
