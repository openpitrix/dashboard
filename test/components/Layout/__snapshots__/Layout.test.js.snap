// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout basic render 1`] = `
<div
  class="layout hasMenu"
>
  <div
    class="nav"
  >
    <ul
      class="topNav"
    >
      <li
        class="navItem firstElem"
      >
        <a
          href="/user/apps"
        >
          <img
            class="icon"
            src="/logo_icon.svg"
          />
        </a>
        <a
          href="/user/apps"
        >
          <label
            class="title"
          >
            QingCloud App Center
          </label>
        </a>
      </li>
    </ul>
    <ul
      class="bottomNav"
    >
      <li>
        <div
          class="tooltip popover iconOuter"
        >
          <div
            class="target"
          >
            <span
              class="icon icon-human icon"
              style="width:20px;height:20px"
            >
              <svg
                class="qicon qicon-human qicon-dark"
              >
                <use
                  href="#qui-human"
                />
              </svg>
            </span>
            <label
              class="title"
            >
              My account
            </label>
          </div>
          <div
            class="popper pi-popover-popper"
            style="position:absolute;pointer-events:none;opacity:0"
          >
            <div>
              <div
                class="pi-popover-content"
              >
                <ul
                  class="menuLayer"
                >
                  <li>
                    <span
                      class="userIcon"
                    >
                      <span
                        class="icon icon-human iconImg"
                        style="width:20px;height:20px"
                      >
                        <svg
                          class="qicon qicon-human qicon-dark"
                        >
                          <use
                            href="#qui-human"
                          />
                        </svg>
                      </span>
                    </span>
                    <span
                      class="username"
                    />
                    <span
                      class="devIconOuter undefined"
                    >
                      <span
                        class="icon icon-enterprise devIcon"
                        style="width:8px;height:8px"
                      >
                        <svg
                          class="qicon qicon-enterprise qicon-white"
                        >
                          <use
                            href="#qui-enterprise"
                          />
                        </svg>
                      </span>
                    </span>
                  </li>
                  <li
                    class="dev"
                  >
                    <a
                      href="/"
                    >
                      <span
                        class="icon icon-appcenter icon"
                        style="width:16px;height:16px"
                      >
                        <svg
                          class="qicon qicon-appcenter qicon-dark"
                        >
                          <use
                            href="#qui-appcenter"
                          />
                        </svg>
                      </span>
                      <span
                        class="label"
                      >
                        App Center
                      </span>
                    </a>
                  </li>
                  <li
                    class=""
                  >
                    <a
                      href="/user/profile/account"
                    >
                      <span
                        class="icon icon-folder icon"
                        style="width:16px;height:16px"
                      >
                        <svg
                          class="qicon qicon-folder qicon-dark"
                        >
                          <use
                            href="#qui-folder"
                          />
                        </svg>
                      </span>
                      <span
                        class="label"
                      >
                        Account Info
                      </span>
                    </a>
                  </li>
                  <li
                    class=""
                  >
                    <a
                      href="/user/profile/password"
                    >
                      <span
                        class="icon icon-lock icon"
                        style="width:16px;height:16px"
                      >
                        <svg
                          class="qicon qicon-lock qicon-dark"
                        >
                          <use
                            href="#qui-lock"
                          />
                        </svg>
                      </span>
                      <span
                        class="label"
                      >
                        Change Password
                      </span>
                    </a>
                  </li>
                  <li
                    class="disabled"
                  >
                    <a
                      href="/"
                    >
                      <span
                        class="icon icon-loudspeaker icon"
                        style="width:16px;height:16px"
                      >
                        <svg
                          class="qicon qicon-loudspeaker qicon-dark"
                        >
                          <use
                            href="#qui-loudspeaker"
                          />
                        </svg>
                      </span>
                      <span
                        class="label"
                      >
                        Notice settings
                      </span>
                    </a>
                  </li>
                  <li
                    class="disabled"
                  >
                    <a
                      href="/"
                    >
                      <span
                        class="icon icon-creditcard icon"
                        style="width:16px;height:16px"
                      >
                        <svg
                          class="qicon qicon-creditcard qicon-dark"
                        >
                          <use
                            href="#qui-creditcard"
                          />
                        </svg>
                      </span>
                      <span
                        class="label"
                      >
                        Payment
                      </span>
                    </a>
                  </li>
                  <li
                    class="divider"
                  >
                    <a
                      href="/logout"
                    >
                      <span
                        class="icon icon-previous icon"
                        style="width:16px;height:16px"
                      >
                        <svg
                          class="qicon qicon-previous qicon-dark"
                        >
                          <use
                            href="#qui-previous"
                          />
                        </svg>
                      </span>
                      <span>
                        Log out
                      </span>
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div
    class=""
    style="max-width:100%"
  >
    <div>
      test
    </div>
  </div>
</div>
`;
